docs/ga-roadmap.md

```diff
@@ -21,6 +21,7 @@ All stabilized modules will conform to the API expectations outlined in the [VER
 Explicitly, the following are not in the scope of v1 for the purposes of this document:
 
 * stabilization of additional components/APIs needed by distribution maintainers. Vendors are not the audience
+  * This explicitly excludes the `service` and `otelcol` modules, for which we will only guarantee that there are no breaking changes impacting end-users of the binary after 1.0, while Go API only changes will continue to be admissible until these modules are tagged as 1.0.
```

Comment:
If we're trying to cut Go API stability from binary stability, are there other modules we could consider?

Reply:
I don't think so because:
1. It would be confusing to release Collector 1.0 without marking the components in that distro as 1.0: end-users identify the version of the Go module with the version of the component.
2. Because of semver, we can't make a 1.0 module A depend on a 0.x module B (if B's types are part of A's public API). So we need to make (effectively) all dependencies of e.g. OTLP receiver 1.0 as well

This only leaves anything 'above' OTLP receiver and friends in the dependency tree as something we can consider out of scope IMO, which boils down to `otelcol` and `service`.

Reply:
> It would be confusing to release Collector 1.0 without marking the components in that distro as 1.0: end-users identify the version of the Go module with the version of the component.

I think we are going to run into the problem of 1.0 binaries/images with 0.x components regardless of any `service`/`otelcol` decision we make. We're covered by https://github.com/open-telemetry/oteps/blob/main/text/0232-maturity-of-otel.md#maturity-levels and the Java Agent already sets precedence for stable binaries/images that include 0.x instrumentation.

Reply:
I mean that we'll run into it for `Core` `Contrib` and `Kubernetes` 1.0 distributions.